src/components/Navbar/Navbar.jsx

```diff
@@ -3,13 +3,31 @@ import logo from "../../assets/ecomm-logo.png";
 import { useNavigate } from "react-router";
 import { useAuth } from "../../context/auth-context";
 import { Link } from "react-router-dom";
+import { useFilter } from "../../context/filter-context";
 import { useCart } from "../../context/cart-context";
-import {useWishlist} from '../../context/wishlist-context'
+import { useWishlist } from "../../context/wishlist-context";
+import { useState } from "react";
+import Search from "../search/Search";
 const Navbar = () => {
   const navigate = useNavigate();
-  const {cartState} = useCart();
-  const {wishlistState} = useWishlist();
+  const { state } = useFilter();
+  const { wishlistState } = useWishlist();
+  const { cartState } = useCart();
   const { currentUser, signout } = useAuth();
+  const [searchResult, setSearchResult] = useState([]);
+  const searchHandler = (e) => {
+    let inputVal = e.target.value.toLowerCase().trim();
+    let inputLen =inputVal.length;
+    let result = state.data.filter(
+      (item) =>
+        item.title.toLowerCase().substring(0,inputLen).includes(inputVal) ||
+        item.categoryName.toLowerCase().substring(0,inputLen).includes(inputVal)||
+        item.brand.toLowerCase().substring(0,inputLen).includes(inputVal)
```

Comment:
Loved the way of writing code, you made it really easy to understand.

Reply:
thank you for the appreciation atul